Demo/Application/Base/AppDelegate.swift

```diff
@@ -0,0 +1,81 @@
+        registerDefaultsFromSettings()
+        BTAppContextSwitcher.sharedInstance.returnURLScheme = BraintreeDemoAppDelegatePaymentsURLScheme
+        
+        UserDefaults.standard.setValue(true, forKey: "magnes.debug.mode")
+        
+        return true
+    }
+    
+    func registerDefaultsFromSettings() {
+        if ProcessInfo.processInfo.arguments.contains("-EnvironmentSandbox") {
+            UserDefaults.standard.set(BraintreeDemoEnvironment.sandbox, forKey: BraintreeDemoSettings.EnvironmentDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-EnvironmentProduction") {
+            UserDefaults.standard.set(BraintreeDemoEnvironment.production, forKey: BraintreeDemoSettings.EnvironmentDefaultsKey)
+        }
+        
+        if ProcessInfo.processInfo.arguments.contains("-ClientToken") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.clientToken, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-TokenizationKey") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.tokenizationKey, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-MockedPayPalTokenizationKey") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.mockedPayPalTokenizationKey, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        } else if ProcessInfo.processInfo.arguments.contains("-UITestHardcodedClientToken") {
+            UserDefaults.standard.set(BraintreeDemoAuthType.uiTestHardcodedClientToken, forKey: BraintreeDemoSettings.AuthorizationTypeDefaultsKey)
+        }
+        
+        UserDefaults.standard.removeObject(forKey: "BraintreeDemoSettingsAuthorizationOverride")
+        ProcessInfo.processInfo.arguments.forEach { arg in
+            if arg.contains("-Integration:") {
+                let testIntegration = arg.replacingOccurrences(of: "-Integration:", with: "")
+                UserDefaults.standard.setValue(testIntegration, forKey: "BraintreeDemoSettingsIntegration")
+            } else if arg.contains("-Authorization:") {
+                let testIntegration = arg.replacingOccurrences(of: "-Authorization:", with: "")
+                UserDefaults.standard.setValue(testIntegration, forKey: "BraintreeDemoSettingsAuthorizationOverride")
+            }
+        }
+        
+        if ProcessInfo.processInfo.arguments.contains("-ClientTokenVersion2") {
+            UserDefaults.standard.set("2", forKey: "BraintreeDemoSettingsClientTokenVersionDefaultsKey")
+        } else if ProcessInfo.processInfo.arguments.contains("-ClientTokenVersion3") {
+            UserDefaults.standard.set("3", forKey: "BraintreeDemoSettingsClientTokenVersionDefaultsKey")
+        }
+        // End checking for testing arguments
```

Comment:
Could we extract setting up the testing arguments into it's own method? We can also probably extract some of these into constants:
```swift
let arguments = ProcessInfo.processInfo.arguments
let userDefaults = UserDefaults.standard
```